app/src/main/java/org/dhis2/usescases/teiDashboard/dashboardfragments/relationships/RelationshipRepositoryImpl.kt

```diff
@@ -384,10 +384,50 @@ class RelationshipRepositoryImpl(
                 }
             }
 
-        return if (attrValuesFromType.isNotEmpty()) {
-            attrValuesFromType
-        } else {
-            listOf(Pair("uid", teiUid))
+        val attrValueFromProgramTrackedEntityAttribute = mutableListOf<Pair<String, String>>()
+        val teiTypeUid = d2.trackedEntityModule().trackedEntityInstances()
+            .uid(teiUid).blockingGet().trackedEntityType()
+        val teiTypeName = d2.trackedEntityModule().trackedEntityTypes()
+            .uid(teiTypeUid).blockingGet().name()!!
```

Comment:
Is this being call outside the main thread?

Reply:
@Balcan

Reply:
Yes, all taht is being called in a IO thread